docs/HowToHashiCorpVault.md

```diff
@@ -0,0 +1,270 @@
+    vault.hashicorp.com/agent-inject: "true"
+    vault.hashicorp.com/agent-inject-secret-ca.cert: nuodb.com/TLS
+    vault.hashicorp.com/agent-inject-secret-nuocmd.pem: nuodb.com/TLS
+    vault.hashicorp.com/agent-inject-template-ca.cert: |
+      {{- with secret "nuodb.com/TLS" -}}
+        {{ .Data.data.tlsCACert | base64Decode }}
+      {{- end }}
+    vault.hashicorp.com/agent-inject-template-nuocmd.pem: |
+      {{- with secret "nuodb.com/TLS" -}}
+        {{ .Data.data.tlsClientPEM | base64Decode }}
+      {{- end }}
+    vault.hashicorp.com/role: nuodb
+    vault.hashicorp.com/secret-volume-path: /etc/nuodb/keys
+```
+
+The NuoDB database engine pod only requires the client credentials (`nuocmd.pem`) and the public CA certificate (`ca.cert`).
+These files are not password protected.
+
+```
+helm install -n nuodb -f vault-annotations-database.yaml database nuodb/database
+```
+
+The NuoDB domain should now consist of three admin tier pods and a TE, and SM pod.
+All pods should contain two init containers and two containers.
+Validate that the pods are ready:
+
+```
+$ kubectl get pods -n nuodb
+NAME                                               READY   STATUS    RESTARTS   AGE
+admin-nuodb-cluster0-0                             2/2     Running   0          10m
+admin-nuodb-cluster0-1                             2/2     Running   0          10m
+admin-nuodb-cluster0-2                             2/2     Running   0          10m
+sm-database-nuodb-cluster0-demo-hotcopy-0          2/2     Running   0          99s
+te-database-nuodb-cluster0-demo-556697c994-z6m5d   2/2     Running   0          99s
+``` 
+
+Validate that the domain is healthy using nuocmd.
+```
+$ kubectl exec -it -n nuodb admin-nuodb-cluster0-0 -c admin -- nuocmd show domain
+server version: 4.0.8-2-881d0e5d44, server license: Community
+server time: 2021-03-03T20:39:31.002, client token: 20f397da114c6e55567ea5d2f53660941f308bea
+Servers:
+  [admin-nuodb-cluster0-0] admin-nuodb-cluster0-0.nuodb.nuodb.svc.cluster.local:48005 [last_ack = 1.27] [member = ADDED] [raft_state = ACTIVE] (LEADER, Leader=admin-nuodb-cluster0-0, log=0/29/29) Connected *
+  [admin-nuodb-cluster0-1] admin-nuodb-cluster0-1.nuodb.nuodb.svc.cluster.local:48005 [last_ack = 1.27] [member = ADDED] [raft_state = ACTIVE] (FOLLOWER, Leader=admin-nuodb-cluster0-0, log=0/29/29) Connected
+  [admin-nuodb-cluster0-2] admin-nuodb-cluster0-2.nuodb.nuodb.svc.cluster.local:48005 [last_ack = 1.27] [member = ADDED] [raft_state = ACTIVE] (FOLLOWER, Leader=admin-nuodb-cluster0-0, log=0/29/29) Connected
+Databases:
+  demo [state = RUNNING]
+    [SM] sm-database-nuodb-cluster0-demo-hotcopy-0/10.1.2.17:48006 [start_id = 0] [server_id = admin-nuodb-cluster0-1] [pid = 118] [node_id = 1] [last_ack =  9.22] MONITORED:RUNNING
+    [TE] te-database-nuodb-cluster0-demo-556697c994-z6m5d/10.1.2.16:48006 [start_id = 1] [server_id = admin-nuodb-cluster0-2] [pid = 44] [node_id = 2] [last_ack =  3.20] MONITORED:RUNNING
+```
```

Comment:
Should we add some steps for troubleshooting in case something is wrong. For example getting the logs of vault init pod, etc?
This is also valid for the test in case of failure.